src/transformers/api-structure-previews.ts

```diff
@@ -1,48 +1,253 @@
+      // Docusaurus could cause that if they limited how many files are being
+      // processed in parallel such that too many docs are awaiting others
+      const timeoutId = setTimeout(() => {
+        // links in translated locale [xy] have their paths prefixed with /xy/
+        const isTranslatedDoc = !relativeStructurePath.startsWith('/docs/');
+
+        if (isTranslatedDoc) {
+          // If we're running locally we might not have translations downloaded
+          // so if we don't find it locally just supply the default locale
+          const [_fullPath, locale, docPath] = relativeStructurePath.match(
+            /\/([a-z][a-z])\/docs\/(.*)/
+          );
+          const defaultLocalePath = `/docs/${docPath}`;
+          const localeDir = path.join(__dirname, '..', '..', 'i18n', locale);
+
+          if (!fs.existsSync(localeDir)) {
+            if (fileContent.has(defaultLocalePath)) {
+              const { promise } = fileContent.get(defaultLocalePath);
+              promise.then((content) => resolve(content));
+              return;
+            }
+          }
+        }
+
+        reject(
+          new Error(
+            `Timed out waiting for API structure content from ${relativeStructurePath}`
+          )
+        );
+      }, 60_000);
+
+      const promise = new Promise<Parent>((resolve_, reject_) => {
+        resolve = (value: Parent) => {
+          clearTimeout(timeoutId);
+          resolve_(value);
+        };
+        reject = reject_;
+      });
+
+      fileContent.set(relativeStructurePath, { promise, resolve });
+    }
+
+    const { promise: targetStructure } = fileContent.get(relativeStructurePath);
+
+    if (toString(node).length > 0) {
+      mutationPromises.add(
+        targetStructure
+          .then((structureContent) => {
+            if (isInline) {
+              // we inline the structure content as the last sibling of the current node
+              const siblings = parents[parents.length - 1].children;
+              const filtered = filter(
+                structureContent,
+                (node) => node.type !== 'heading'
+              );
+              siblings.push(filtered);
+            } else {
+              const HAST = toHast(structureContent as Root, {
+                handlers: {
+                  mdxJsxFlowElement: (_, node) => {
+                    if (
+                      node.name === 'APIStructurePreview' &&
+                      node?.data?._originalLink
+                    ) {
+                      const { href, text } = node.data._originalLink;
+                      return h('a', { href }, [text]);
+                    } else if (node.name === 'header') {
+                      const text = toString(node);
+                      return h('h1', [text]);
+                    }
+                  },
+                },
+              });
+
+              const sanitized = sanitize(HAST, {
```

Comment:
Let's add a more detailed comment before this line.

Reply:
https://github.com/electron/website/pull/690/commits/c3d0a0eb3522c14c356049aa011af8688cee800a